retriever/__main__.py

```diff
@@ -138,15 +138,36 @@ def main():
                 from retriever import lscolumns
                 lscolumns.printls(dataset_names())
             else:
-                all_scripts = datasets(args.l[0])
-                print("Available datasets : {}\n".format(len(all_scripts)))
-                count = 1
-                for script in all_scripts:
-                    print("{}. {}".format(count, script.title))
-                    print(script.name)
-                    print(script.keywords)
-                    print()
-                    count += 1
+                if len(args.l) > 0:
+                    all_license = dataset_licenses()
+                    # filter valid license names. if there is no valid license then its keywords
+                    licenses = [l for l in args.l if l in all_license]
+                    if len(licenses) > 0:
+                        # search by license
+                        license_scripts = datasets(None, licenses)
```

Comment:
So as mentioned above, at this point we should return both the lists of scripts(datasets) that have either a keyword that is close to (MIT or GPL) , or a  licence that are exactly (MIT or GPL)